pybamm/expression_tree/symbol.py

```diff
@@ -361,28 +380,32 @@ def is_constant(self):
         # do the search, return true if no relevent nodes are found
         return all([not (isinstance(n, search_types)) for n in self.pre_order()])
 
-    def evaluates_to_value(self, value):
+    def evaluate_ignoring_errors(self):
         """
-        Returns True if evaluating the expression returns a given constant value.
-        Returns False otherwise, including if NotImplementedError or TyperError
-        is raised.
-        !Not to be confused with isinstance(self, pybamm.Scalar)!
-
-        Parameters
-        ----------
-        value: Number
-            the value to compare against
+        Evaluates the expression. If a node exists in the tree that cannot be evaluated
+        as a scalar or vectr (e.g. Parameter, Variable, StateVector), then None is
+        returned. Otherwise the result of the evaluation is given
 
         See Also
         --------
         evaluate : evaluate the expression
 
         """
-        return (
-            self.is_constant()
-            and self.evaluates_to_number()
-            and self.evaluate() == value
-        )
+        try:
+            result = self.evaluate(t=0)
```

Comment:
I know I started this but it now makes me a bit uncomfortable (avoiding the `t` dependence). Works for now and I can't think of a better way but something to keep an eye out for

Reply:
yea, me too :) For the simplifications, I always use is_constant(), so I should be able to remove the `t=0`, but it would be needed for `evaluates_to_number`.